src/clang-c-frontend/symbolic_types.cpp

```diff
@@ -1,63 +1,37 @@
 #include "symbolic_types.h"
 
-void get_complete_struct_type(struct_typet &type, const namespacet &ns)
+/* TODO: this is basically thrash_type_symbols() modulo pointers in slow */
+bool get_complete_type(typet &type, const namespacet &ns)
 {
-  for(auto &comp : type.components())
+  if(type.is_struct() || type.is_union())
   {
-    typet &it_type = comp.type();
-    if(it_type.is_symbol())
-    {
-      const symbolt *complete_type = ns.lookup(it_type.identifier());
-      assert(complete_type);
-      if(complete_type->type.is_struct())
-      {
-        it_type = complete_type->type;
-        // follow the type recursively to replace all symbolic types
-        get_complete_struct_type(to_struct_type(it_type), ns);
-      }
-    }
+    bool r = false;
+    for(auto &comp : to_struct_union_type(type).components())
+      r |= get_complete_type(comp.type(), ns);
+    return r;
   }
+  else if(type.is_array())
+  {
+    return get_complete_type(type.subtype(), ns);
+  }
+  else if(type.is_symbol())
+  {
+    const symbolt *complete_type = ns.lookup(type.identifier());
+    assert(complete_type);
+    assert(complete_type->is_type);
+    assert(!complete_type->type.is_symbol());
+    type = complete_type->type;
+    get_complete_type(type, ns);
+    return true;
+  }
+  return false;
 }
 
 bool contains_symbolic_struct_types(
```

Comment:
can we get rid of this method and use `get_complet_type` directly? I think this method is only used in one place anyway.

Reply:
Done.